src/server/worker/master.go

```diff
@@ -212,15 +212,15 @@ nextInput:
 			var jobID string
 			var jobInfo pps.JobInfo
 			ok, err := jobIter.Next(&jobID, &jobInfo)
-			if err != nil {
+			if err != nil && !col.IsErrNotFound(err) {
 				return err
 			}
-			if !ok {
+			if !ok || col.IsErrNotFound(err) {
 				ok, err := oldJobIter.Next(&jobID, &jobInfo)
-				if err != nil {
+				if err != nil && !col.IsErrNotFound(err) {
 					return err
 				}
-				if !ok {
+				if !ok || col.IsErrNotFound(err) {
```

Comment:
Would it make sense to put this at the collection level rather than this level. It's pretty conterintuitive to need to handle not found errors when you using an iterator. So maybe we should just make it impossible as part of the api. Would probably insulate us from other potential bugs and also make this code a bit less confusing. At the very least we should have a comment here about why we need to handle not found errors during iteration.

Reply:
That makes sense.